### DISCUSSION TITLE
We need better UX for directories in search

### DISCUSSION THREAD

**Discussion by natepisarski** [Kong/insomnia#6942] [Ideas] [answered]
I wanted to share my thoughts about a search regression that took place with version 8. This 1 change has prevented me from upgrading Insomnia, and I'm sure many other developers are unhappy with it too. The issue is 2-fold:

* When you type in the search bar, requests that match the query **are not organized by directory** in the search results.
* When you type the name of a folder, **its contents are not shown**

Both of these are recent issues, starting with (I assume) 8.0. For instance, look at this search on `Insomnia 2022.6.0` (the version I will use until this is resolved):

### Organization by directory
The directory structure in these screenshots looks like this:

* Customers
  * Get Customers
  * Get Metadata
* Users
  * Get Users
  * Get Metadata

**Pre-8.0**  
![image](https://github.com/Kong/insomnia/assets/5361699/c565f214-edaa-4696-945b-2c6626c4d7c3)

**Post-8.0**
![image](https://github.com/Kong/insomnia/assets/5361699/f56c9cd0-7778-4e0d-9b98-9082cc387e87)

### Displaying a folder
**Pre-8.0**  
![image](https://github.com/Kong/insomnia/assets/5361699/30d40ec1-3c70-44b0-aad8-d063b564da82)

**Post-8.0**  
![image](https://github.com/Kong/insomnia/assets/5361699/b8b9eed4-3f85-4751-8c8b-bcdf2c049fd4)

--------

This UX is obviously problematic, and becomes exponentially more so when you have collections with either hundreds or thousands of requests. If you have a lot of requests named the same thing (`Get Metadata`) clicking the correct one becomes a game of chance.

**Comment by subnetmarco** [comment, marked as answer]
This is not the first time I hear about this, so I prioritized it for the January release, let's see if we can figure out why this is happening.
